components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4393,4 +4402,84 @@ public static String resolveTenantDomain(HttpServletRequest request) {
         }
         return IdentityTenantUtil.getTenantDomainFromContext();
     }
+
+    /**
+     * Checks if the password had expired.
+     *
+     * @param tenantDomain        The tenant domain of the user trying to authenticate
+     * @param tenantAwareUsername The tenant aware username of the user trying to authenticate
+     * @return True if the password had expired
+     * @throws IdentityOAuth2Exception if the authentication failed for the user trying to login
+     */
+    public static boolean hadPasswordExpired(String tenantDomain, String tenantAwareUsername)
```

Comment:
There should be a new line after line number 4414. Let's use WSO2 styling configs.